### PR TITLE
Make "ThreadStateSliceInfo" a class (rather than a proto)

### PR DESCRIPTION
**Pull request by florian-kuebler**
Test: Compile
Bug: http://b/226551541

### CODE REVIEW

File: src/ClientData/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,7 @@ target_sources(ClientData PUBLIC
         include/ClientData/TimerDataInterface.h
         include/ClientData/TimerDataManager.h
         include/ClientData/TimestampIntervalSet.h
+        include/ClientData/ThreadStateSliceInfo.h
```
Comment (dpallotti) [inline src/ClientData/CMakeLists.txt:39]: Move up to keep the alphabetical order.

File: src/ClientData/include/ClientData/ThreadStateSliceInfo.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef CLIENT_DATA_THREAD_STATE_SLICE_INFO_H_
```
Comment (dpallotti) [inline src/ClientData/include/ClientData/ThreadStateSliceInfo.h:4]:
```suggestion
// found in the LICENSE file.

#ifndef CLIENT_DATA_THREAD_STATE_SLICE_INFO_H_
```

File: src/CaptureClient/CaptureEventProcessorTest.cpp [hunk 3]
```diff
@@ -1169,29 +1169,32 @@ TEST(CaptureEventProcessor, CanHandleThreadStateSlices) {
   event_processor->ProcessEvent(runnable_event);
   event_processor->ProcessEvent(dead_event);
 
+  ASSERT_TRUE(actual_running_thread_state_slice_info.has_value());
+  ASSERT_TRUE(actual_runnable_thread_state_slice_info.has_value());
+  ASSERT_TRUE(actual_dead_thread_state_slice_info.has_value());
```
Comment (dpallotti) [inline src/CaptureClient/CaptureEventProcessorTest.cpp:1174]: Personally I would have placed each of these right before the corresponding `EXPECT_EQ`s below.

**Review by dpallotti — APPROVED** [review]
Thanks, super minor comments.

File: src/ClientData/CMakeLists.txt (hunk 1 above)
Comment (florian-kuebler) [reply src/ClientData/CMakeLists.txt:39]: Ups. Done.

File: src/ClientData/include/ClientData/ThreadStateSliceInfo.h (hunk 2 above)
Comment (florian-kuebler) [reply src/ClientData/include/ClientData/ThreadStateSliceInfo.h:4]: Done.

File: src/CaptureClient/CaptureEventProcessorTest.cpp (hunk 3 above)
Comment (florian-kuebler) [reply src/CaptureClient/CaptureEventProcessorTest.cpp:1174]: Done.

**Pull request merged**